extensions/amp-form/amp-form.md

```diff
@@ -291,6 +296,38 @@ This shows all validation errors on all invalid inputs when the form is submitte
 #### As You Go
 This allows your user to see validation messages as they're interacting with the input, if the email they typed is invalid they'll see the error right away and once fixed the error goes away.
 
+## Variable Substitutions
+__(<a href="https://www.ampproject.org/docs/reference/experimental.html">experimental</a>)__
+
+`amp-form` allows [variable substitutions](../../spec/amp-var-substitutions.md) for inside hidden inputs with `default-value` attribute. On each submission, `amp-form` finds all `input[type=hidden]` inside the form and apply variable substitutions to `default-value` attribute and fill the result in the input `value`.
```

Comment:
I am happy to change this, `default-value` was a quick suggestion from @cramforce and I took it and ran with it 😄. 

> (a) Is using data- attributes a better approach here. (I'm not sure where to use them and when not.)

Me and you both. I am never sure whether to choose to go with `data-` or not. I'd love for us to decide on a better way to choose and set of rules to follow when evaluating attributes naming.

> (b) "default" suggests to me something that's defined (in fact, it's the value used if another one doesn't come along to replace it). Here, however, what's really being constructed is a tokenized string. I wonder if we can make the attribute name clearer.

We could rename `default-value` to `resolve-value` or something clearer, value-for-substitution also sounds weird but might be clearer.